ReSharper.FSharp/src/FSharp/FSharp.Psi.Intentions/src/Intentions/FunctionAnnotationAction.fs

```diff
@@ -154,19 +165,26 @@ module SpecifyTypes =
             ParenPatUtil.addParens listConsParenPat |> ignore
 
     let private specifyParameterTypes (decl: IParameterOwnerMemberDeclaration) (mfv: FSharpMemberOrFunctionOrValue) displayContext =
-        let types = getFunctionTypeArgs true mfv.FullType
-        let parameters = decl.ParametersDeclarations |> Seq.map _.Pattern
-
-        let rec specifyParameterTypes (types: FSharpType seq) (parameters: IFSharpPattern seq) isTopLevel =
-            for fcsType, parameter in Seq.zip types parameters do
+        let rec specifyParameterTypes (fcsParamGroups: 'a seq) (getFcsType: 'a -> FSharpType) (enumerate: 'a -> 'a seq) (parameters: IFSharpPattern seq) isTopLevel =
+            for fcsParamGroup, parameter in Seq.zip fcsParamGroups parameters do
                 match parameter.IgnoreInnerParens() with
                 | :? IConstPat | :? ITypedPat -> ()
                 | TupleLikePattern pat when isTopLevel ->
-                    specifyParameterTypes fcsType.GenericArguments pat.Patterns false
+                    let fcsTypes = enumerate fcsParamGroup
+                    specifyParameterTypes fcsTypes getFcsType enumerate pat.Patterns false
                 | pattern ->
+                    let fcsType = getFcsType fcsParamGroup
                     specifyPatternType displayContext fcsType pattern
 
-        specifyParameterTypes types parameters true
+        let parameters = decl.ParametersDeclarations |> Seq.map _.Pattern
+
+        if mfv.IsMember then
+            let enumerate x = x |> Seq.map (fun x -> [|x|] :> IList<_>)
+            let types = mfv.CurriedParameterGroups
+            specifyParameterTypes types (fun x -> x[0].Type) enumerate parameters true
+        else
+            let types = getFunctionTypeArgs true mfv.FullType
+            specifyParameterTypes types id (_.GenericArguments) parameters true
```

Comment:
Are the parens required here?

Reply:
We are currently using .NET 8.0.100 for GitHub Actions, which doesn't support shorthand syntax without parens properly. 
We can upgrade it slightly in a separate PR or leave the brackets here.